app/v1/services/RetrieveBusinessDetailsService.scala

```diff
@@ -31,14 +32,16 @@ import scala.concurrent.{ExecutionContext, Future}
 @Singleton
 class RetrieveBusinessDetailsService @Inject() (connector: RetrieveBusinessDetailsConnector) extends BaseService {
 
-  def retrieveBusinessDetailsService(request: RetrieveBusinessDetailsRequest)(implicit
+  def retrieveBusinessDetailsService(request: RetrieveBusinessDetailsRequest, appConfig: AppConfig)(implicit
       ctx: RequestContext,
       ec: ExecutionContext): Future[ServiceOutcome[RetrieveBusinessDetailsResponse]] = {
 
+    val r10AdditionalFieldsEnabled = FeatureSwitches()(appConfig).r10AdditionalFieldsEnabled
```

Comment:
cant we use featureSwitches. r10AdditionalFieldsEnabled here instead of creating new? May be i am overlooking something?